Automattic/jetpack
# Replace "Site visibility" with a link to Calypso

Related to https://github.com/Automattic/dotcom-forge/issues/6694, https://github.com/Automattic/wpcomsh/pull/1893

## Proposed changes:


This PR replaces "Site visibility" with a link to Calypso on the Reading Settings page in wp-admin (i.e., This PR fixes an unexpected behavior: the Site Visibility setting is shown in wp-admin of Atomic Classic). 

### before
- Simple Classic
	- <img width="960" alt="Screenshot 2024-05-31 at 13 54 38" src="https://github.com/Automattic/jetpack/assets/5287479/2db751de-7b3b-456f-8dcb-4c8181335f20">
- Atomic Classic
	- <img width="975" alt="Screenshot 2024-05-31 at 13 55 55" src="https://github.com/Automattic/jetpack/assets/5287479/a271155c-c81e-4816-9dd1-be6dd6c41759">

### after
- Simple Classic, Atomic Classic 
	- <img width="489" alt="Screenshot 2024-05-31 at 13 54 02" src="https://github.com/Automattic/jetpack/assets/5287479/ff12a6e1-455d-48d2-b2f2-07c4584b83ef">
- Atomic Classic without Jetpack connected
	- private site
		- <img width="889" alt="Screenshot 2024-05-31 at 15 36 44" src="https://github.com/Automattic/jetpack/assets/5287479/f37109c4-64dd-4cb3-8736-651f0d2c64d8">
	- public site
		- <img width="1012" alt="Screenshot 2024-05-31 at 15 55 14" src="https://github.com/Automattic/jetpack/assets/5287479/80739847-8f82-423a-a0b1-7e7490dbc956">

---
**The Site Visibility setting should just be a link to Calypso**. See https://github.com/Automattic/dotcom-forge/issues/6526#issuecomment-2116939266 for more details. 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Atomic Classic

- Prepare an Atomic Classic site with the flag `wpcom_classic_early_release=1`
- Patch this PR
- Go to Settings > Reading
	- See `Site visibility > Manage your site visibility settings`
- Click `Manage your site visibility settings` and set your site `Coming Soon` or `Private`
	- Go to Jetpack > My Jetpack
	- Disconnect Jetpack from the bottom of the page
	- Go to Settings > Reading
	- See `Site visibility > Jetpack is disconnected & site is private. Reconnect Jetpack to manage site visibility settings.`
- Go to Jetpack > My Jetpack
	- Reconnect Jetpack
	- Go to Settings > Reading
	- Click `Manage your site visibility settings` and set your site `Public`
	- Go to Jetpack > My Jetpack
	- Disconnect Jetpack from the bottom of the page
	- Go to Settings > Reading
	- See the setting as the same as Core
	- Confirm the option can be updated

Simple Classic

- Prepare an Atomic Classic site with the flag `wpcom_classic_early_release=1`
- Patch this PR
- Go to Settings > Reading
- See `Site visibility > Manage your site visibility settings`
